docs/source/error-codes/self-service/index.rst

```diff
@@ -7,6 +7,138 @@ Self-Service Error Codes (Experimental)
 .. toctree::
    :hidden:
 
+Overview
+*********
+
+
+TODO: Failure vs. error.
+
+.. _gRPC: https://grpc.github.io/grpc/core/md_doc_statuscodes.html
+
+Participant API server has been using a combination of gRPC_
+status codes and custom messages to signal erroneous or unexpected conditions.
+This approach remains unchanged in principle while self-service error codes aim at
+enhancing it by providing:
+
+- improved consistency of the returned errors across API endpoints,
+
+- richer message format with clearly distinguished machine readable parts to facilitate
+  automated error handling strategies,
+
+- complete inventory of all error codes with an explanation, suggested resolution and
+  more (as documented in this section).
+
+
+The goal is to enable users, developers and operators to act upon encountered
+errors in a self-service manner such that the need to escalate the problem
+or file a ticket occurs occurs infrequently.
+
+Feature flag
+************
+You can enable self-service error-codes by specifying ``--use-self-service-error-codes``
+from command line.
+
+By default self-service error codes are turned off.
+
+
+Anatomy of an error code
+*****************************
+
+TODO Error categories:
+
+ - what they are,
+
+ - how to make use of them,
+
+ - provide a list of all error categories.
+
+Error message
+===================
+.. code-block:: java
+
+    <ERROR_CODE_ID>(<CATEGORY_ID>,<CORRELATION_ID>:<HUMAN_READABLE_MESSAGE>
+
+The constituent parts are:
+
+  - ``<ERROR_CODE_ID>`` - a unique non empty string containing at most 63 characters:
+    upper-cased letters, underscores or digits.
+  - ``<CATEGORY_ID>`` - a small integer identifying the corresponding error categories.
+  - ``<CORRELATION_ID>`` - a string helpful at identifying originating request.
+    ``0`` if no correlation id is given. TODO: Are there limits to its size?
```

Comment:
We take this value from the submission-id (where provided). The limit is thus the same as for submission-id. We only say there that it must be a valid ledger string and provide a regex for it no mention of the size. Typically this will be a UUID